### PR TITLE
feat: add learners and enrollments tab component

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2481](https://openedx.atlassian.net/browse/PROD-2481)
Added a new component separating components in tabs. 

Screenshot 2021-09-24 at 3 48 00 PM

Screenshot 2021-09-24 at 3 48 09 PM

Screenshot 2021-09-24 at 3 47 55 PM

### CODE REVIEW

File: src/users/v2/LearnersAndEnrollments.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Tabs, Tab } from '@edx/paragon';
+import UserSummary from '../UserSummary';
+import EnrollmentsV2 from '../enrollments/v2/Enrollments';
+import SingleSignOnRecords from './SingleSignOnRecords';
+import Licenses from '../licenses/Licenses';
+import EntitlementsV2 from '../entitlements/v2/Entitlements';
+
+export default function LearnersAndEnrollments({
```
Comment (DawoudSheraz) [inline src/users/v2/LearnersAndEnrollments.jsx]: the component name can be improved, such as LearnerInformation, LearnerRecords etc. There are many things besides enrollments, so the name does not specify the component's exact usage.

File: src/users/v2/LearnersAndEnrollments.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    >
+      <Tab eventKey="account" title="Account Information">
+        <br />
+        <UserSummary
+          userData={user}
+          changeHandler={changeHandler}
+        />
+        <Licenses
+          userEmail={user.email}
+        />
```
Comment (DawoudSheraz) [inline src/users/v2/LearnersAndEnrollments.jsx]: move license manager details with SSO tab. Rename the tab to SSO/License Info

**Review by DawoudSheraz — COMMENTED** [review]
We might need to change the tab's style. Overall, looks good.

**Comment by azanbinzahid** [comment]
Please update SS in PR.

File: src/users/v2/LearnerInformation.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Tabs, Tab } from '@edx/paragon';
+import UserSummary from '../UserSummary';
+import EnrollmentsV2 from '../enrollments/v2/Enrollments';
+import SingleSignOnRecords from './SingleSignOnRecords';
+import Licenses from '../licenses/Licenses';
+import EntitlementsV2 from '../entitlements/v2/Entitlements';
+
+export default function LearnersAndEnrollments({
```
Comment (DawoudSheraz) [inline src/users/v2/LearnerInformation.jsx]: change the component name too. Keep file and component name consistent.

File: src/overrides.scss [hunk 4]
```diff
@@ -1,6 +1,16 @@
 // Reverting the container width to 1440px to keep the support tools styling consistent
 // See https://github.com/edx/paragon/pull/533 for paragon breaking change
 
+.learners-and-enrollments {
```
Comment (DawoudSheraz) [inline src/overrides.scss]: update the class name, learner-record-tab etc.

File: src/users/v2/LearnerInformation.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      <br />
+      <Tabs
+        id="learners-and-enrollments"
+        className="learners-and-enrollments"
+      >
+        <Tab eventKey="account" title="Account Information">
+          <br />
+          <UserSummary
+            userData={user}
+            changeHandler={changeHandler}
```
Comment (DawoudSheraz) [inline src/users/v2/LearnerInformation.jsx:24]: for note purposes, we will need to remove the top-level changeHandler for summary in UserSummaryV2. Right now, the trigger of this handler reloads the entire user component.

File: src/users/v2/LearnerInformation.test.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+);
+
+describe('Learners and Enrollments component', () => {
+  let wrapper;
+  const props = {
+    user: UserSummaryData.userData,
+    changeHandler: jest.fn(() => {}),
+  };
+
+  beforeEach(async () => {
```
Comment (DawoudSheraz) [inline src/users/v2/LearnerInformation.test.jsx:29]: add a corresponding afterEach to unmount component.

**Review by DawoudSheraz — APPROVED** [review]
nits/refactors to add before merge

File: src/users/v2/LearnerInformation.test.jsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    expect(tabs.at(0).html()).not.toEqual(expect.stringContaining('active'));
+    expect(tabs.at(1).html()).toEqual(expect.stringContaining('active'));
+    expect(tabs.at(2).html()).not.toEqual(expect.stringContaining('active'));
+
+    expect(tabs.at(2).text()).toEqual('SSO/License Info');
+    tabs.at(2).simulate('click');
+    tabs = wrapper.find('nav.nav-tabs a');
+    expect(tabs.at(0).html()).not.toEqual(expect.stringContaining('active'));
+    expect(tabs.at(1).html()).not.toEqual(expect.stringContaining('active'));
+    expect(tabs.at(2).html()).toEqual(expect.stringContaining('active'));
```
Comment (DawoudSheraz) [inline src/users/v2/LearnerInformation.test.jsx]: another thing to add here is that correct components are displayed under each tab. i.e SSO is not present in Entitlement etc. So check that each correct component is present in each tab and the others are not. Split this test into 3 tests, one for each tab.

File: src/users/v2/LearnerInformation.jsx (hunk 5 above)
Comment (Ali-D-Akbar) [reply src/users/v2/LearnerInformation.jsx:24]: Yes, this must be addressed while redesigning UserSummary

**Review by azanbinzahid — APPROVED** [review]

File: src/users/v2/LearnerInformation.jsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+export default function LearnerInformation({
+  user, changeHandler,
+}) {
+  return (
+    <>
+      <br />
+      <Tabs
+        id="learner-information"
+        className="learners-and-enrollments"
```
Comment (DawoudSheraz) [inline src/users/v2/LearnerInformation.jsx]: update the classname here.

File: src/users/v2/LearnerInformation.test.jsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    let tabs = wrapper.find('nav.nav-tabs a');
+
+    tabs.at(1).simulate('click');
+    tabs = wrapper.find('nav.nav-tabs a');
+    expect(tabs.at(0).html()).not.toEqual(expect.stringContaining('active'));
+    expect(tabs.at(1).html()).toEqual(expect.stringContaining('active'));
+    expect(tabs.at(2).html()).not.toEqual(expect.stringContaining('active'));
+
+    expect(tabs.at(0).find('Entitlement')).toBeTruthy();
+    expect(tabs.at(0).find('Enrollment')).toBeTruthy();
```
Comment (DawoudSheraz) [inline src/users/v2/LearnerInformation.test.jsx]: Wait, aren't component names EntitlementV2 and EnrollmentV2? How did this test pass?

**Pull request merged**